src/states/node.rs

```diff
@@ -1426,6 +1426,17 @@ impl Node {
             }
         }
 
+        let acting_as_proxy = if let Authority::Client { ref proxy_node_name, .. } =
+                                     routing_msg.src {
+            proxy_node_name == self.name()
+        } else {
```

Comment:
Sure, that's fine, too.
(I agree with the self-documenting, but in this case I feel it inflates the code a lot. Anyway, I'm okay with just moving the `!acting_as_proxy` to the front.)